clients/src/main/java/org/apache/kafka/common/record/ByteBufferLogInputStream.java

```diff
@@ -62,13 +54,31 @@ public MutableRecordBatch nextBatch() throws IOException {
         batchSlice.limit(batchSize);
         buffer.position(buffer.position() + batchSize);
 
-        if (magic < 0 || magic > RecordBatch.CURRENT_MAGIC_VALUE)
-            throw new CorruptRecordException("Invalid magic found in record: " + magic);
-
         if (magic > RecordBatch.MAGIC_VALUE_V1)
             return new DefaultRecordBatch(batchSlice);
         else
             return new AbstractLegacyRecordBatch.ByteBufferLegacyRecordBatch(batchSlice);
     }
 
+    int nextBatchSize() throws CorruptRecordException {
+        int remaining = buffer.remaining();
+        if (remaining < LOG_OVERHEAD)
+            throw new IllegalStateException("Buffer does not contain batch header");
```

Comment:
This should be `CorruptRecordException` too.

Reply:
We only ever invoke this method from `nextBatch()` which has already done this check, or when growing log cleaner buffers beyond `max.message.size`. So we don't expect this check to fail, hence `IllegalStateException` seems better?

Reply:
We also invoke it from `MemoryRecords`, no?

Reply:
Right, so it seems a bit brittle to assume that some checks have already been done.

Reply:
Since we have the check in `nextBatch` already, would it make sense to grab the record size there and pass it to this function as an argument?

Reply:
I changed the method to return null if there isn't enough data in the buffer, making it consistent with `nextBatch`. Added comments and test.